### PR TITLE
Add Native asset transfer support

### PR DESCRIPTION
**Pull request by shubhampatwa**

### CODE REVIEW

File: src/sync/events/utils.ts [hunk 1]
```diff
@@ -17,32 +23,54 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
...
-              gasPrice,
-              gasUsed,
-              gasFee,
-            };
-          });
+              const gasPrice = tx.gasPrice?.toString();
+              const gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
+              const gasFee = gasPrice && gasUsed ? bn(gasPrice).mul(gasUsed).toString() : undefined;
+              if (!bn(tx.value).isZero()) {
+                onChainTransaction.push({
```
Comment (ptsayli) [inline src/sync/events/utils.ts]: All transactions are on chain transactions, change variable.

File: src/sync/events/utils.ts [hunk 2]
```diff
@@ -17,32 +23,54 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
...
+                gasPrice,
+                gasUsed,
+                gasFee,
+              };
+            }
+          );
           // Save all transactions within the block
           await saveTransactions(chainId, transactions);
 
+          await processOnChainData(chainId, { ftTransferEvents: onChainTransaction });
```
Comment (ptsayli) [inline src/sync/events/utils.ts]: I think we don't need to go through `ft-transfer-events` here. We can directly make an object and add it to the `transferActivity` queue.
Please check if I'm thinking right. @shubhampatwa

**Review by ptsayli — CHANGES_REQUESTED** [review]

File: src/sync/events/utils.ts (hunk 2 above)
Comment (shubhampatwa) [reply src/sync/events/utils.ts]: yeah exactly we need to do that, Need clarity on batchIndex
@ptsayli

File: src/sync/events/utils.ts [hunk 3]
```diff
@@ -17,32 +23,54 @@ export const fetchBlock = async (chainId: number, blockNumber: number, force = f
...
-              blockTimestamp: block.timestamp,
-              gasPrice,
-              gasUsed,
-              gasFee,
-            };
-          });
+              const gasPrice = tx.gasPrice?.toString();
+              const gasUsed = rawTx?.gas ? bn(rawTx.gas).toString() : undefined;
+              const gasFee = gasPrice && gasUsed ? bn(gasPrice).mul(gasUsed).toString() : undefined;
+              if (!bn(tx.value).isZero()) {
```
Comment (shubhampatwa) [inline src/sync/events/utils.ts:38]: @spunkypanda check here

Comment (spunkypanda) [reply src/sync/events/utils.ts:38]: Thanks

**Review by spunkypanda — APPROVED** [review]

**Comment by ptsayli** [comment]
https://github.com/priori-capital/wallet-indexer/pull/4/files#diff-c1fa3b6bc0bc52056ea900ead878b3706dafbe3bfc81deac8d7c86ec21b06b20R110
As we are storing native assets also, We need to create a new event kind. @shubhampatwa

**Comment by shubhampatwa** [comment]
> https://github.com/priori-capital/wallet-indexer/pull/4/files#diff-c1fa3b6bc0bc52056ea900ead878b3706dafbe3bfc81deac8d7c86ec21b06b20R110 As we are storing native assets also, We need to create a new event kind. @shubhampatwa

If I create a new event kind, I will call the same function only to save in user activity, as we are not storing the kind column, do we need to create a new event kind then also? @ptsayli

**Review by ptsayli — APPROVED** [review]
Perfect

**Pull request merged**